stdlib/public/runtime/DynamicCast.cpp

```diff
@@ -764,14 +770,18 @@ tryCastToAnyHashable(
   assert(cast<StructMetadata>(destType)->Description
          == &STRUCT_TYPE_DESCR_SYM(s11AnyHashable));
 
+  bool useCachedNSStringConformance = false;
```

Comment:
I'm not a fan of setting a boolean to test later on in the same code.
I suggest refactoring this with
```
  const HashableWitnessTable *hashableConformance = nullptr;
```
here, then set it along different paths.